pkg/apis/core/validation/validation_test.go

```diff
@@ -7753,88 +7748,110 @@ func TestValidateResizePolicy(t *testing.T) {
 			PodRestartPolicy: "Never",
 		},
 		"ValidMemoryCPUPolicyWithPodRestartPolicy": {
 			PolicyList: []core.ContainerResizePolicy{
-				{ResourceName: "cpu", RestartPolicy: "NotRequired"},
-				{ResourceName: "memory", RestartPolicy: "NotRequired"},
+				{ResourceName: "cpu", RestartPolicy: "PreferNoRestart"},
+				{ResourceName: "memory", RestartPolicy: "PreferNoRestart"},
 			},
-			ExpectError:      false,
 			Errors:           nil,
 			PodRestartPolicy: "Never",
 		},
 	}
+	// Mixin deprecated test cases.
+	const notRequiredCaseSuffix = "(NotRequired)"
 	for k, v := range testCases {
-		errs := validateResizePolicy(v.PolicyList, field.NewPath("field"), &v.PodRestartPolicy)
-		if !v.ExpectError && len(errs) > 0 {
-			t.Errorf("Testcase %s - expected success, got error: %+v", k, errs)
+		if strings.HasSuffix(k, notRequiredCaseSuffix) {
+			continue // Since we're modifying the map while iterating, we may see a case we added. Just skip it.
 		}
-		if v.ExpectError {
-			if len(errs) == 0 {
-				t.Errorf("Testcase %s - expected error, got success", k)
-			}
-			delta := cmp.Diff(errs, v.Errors)
-			if delta != "" {
-				t.Errorf("Testcase %s - expected errors '%v', got '%v', diff: '%v'", k, v.Errors, errs, delta)
+		notRequiredPolicies := []core.ContainerResizePolicy{}
+		hadPreferNoRestart := false
+		for _, p := range v.PolicyList {
+			if p.RestartPolicy == core.ResizeRestartPolicyPreferNoRestart {
+				p.RestartPolicy = core.ResizeRestartPolicyNotRequired
+				hadPreferNoRestart = true
 			}
+			notRequiredPolicies = append(notRequiredPolicies, p)
+		}
+		if hadPreferNoRestart {
+			v.PolicyList = notRequiredPolicies
+			testCases[k+notRequiredCaseSuffix] = v
 		}
 	}
+	for k, v := range testCases {
+		t.Run(k, func(t *testing.T) {
+			opts := PodValidationOptions{AllowResizeRestartPolicyPreferNoRestart: true}
+			errs := validateResizePolicy(v.PolicyList, field.NewPath("field"), &v.PodRestartPolicy, opts)
+			if len(v.Errors) == 0 {
+				assert.NoError(t, errs.ToAggregate())
+			} else {
+				assert.Equal(t, v.Errors, errs)
+			}
+		})
+	}
+
+	for policy := range supportedResizePolicies {
+		t.Run(fmt.Sprintf("AllowResizeRestartPolicyPreferNoRestart=false/policy=%s", policy), func(t *testing.T) {
+			policyList := []core.ContainerResizePolicy{{ResourceName: "cpu", RestartPolicy: policy}}
+			opts := PodValidationOptions{}
+			errs := validateResizePolicy(policyList, field.NewPath("field"), ptr.To(core.RestartPolicyAlways), opts)
+			if policy == core.ResizeRestartPolicyPreferNoRestart {
+				expected := field.ErrorList{field.Forbidden(field.NewPath("field"), `resize policy "PreferNoRestart" requires the "InPlacePodVerticalScalingPreferNoRestart" feature gate`)}
```

Comment:
If the test depends on the gate it should set the gate, so when it changes default the test keeps working (and being useful)